tests/test_user.py

```diff
@@ -1,4 +1,134 @@
-from app.routers.user import create_user, does_user_exist, get_users
+from datetime import datetime
+import pytest
+
+from app.routers.user import (
+    create_user, does_user_exist, get_users
+)
+from app.internal.user.availability import disable, enable
+from app.internal.utils import save
+from app.database.models import UserEvent, Event
+from app.routers.event import create_event
+
+# -----------------------------------------------------
+# Fixtures
+# -----------------------------------------------------
+
+
+@pytest.fixture
+def user1(session):
+    # a user made for testing who doesn't own any event.
+    user = create_user(
+        session=session,
+        username='new_test_username',
+        password='new_test_password',
+        email='new2_test.email@gmail.com',
+        language_id='english'
+    )
+
+    return user
+
+
+@pytest.fixture
+def user2(session):
+    # a user made for testing who already owns an event.
+    user = create_user(
+        session=session,
+        username='new_test_username2',
+        password='new_test_password2',
+        email='new_test_love231.email@gmail.com',
+        language_id='english'
+    )
+
+    data = {
+        'title': 'user2 event',
+        'start': datetime.strptime('2021-05-05 14:59', '%Y-%m-%d %H:%M'),
+        'end': datetime.strptime('2021-05-05 15:01', '%Y-%m-%d %H:%M'),
+        'location': 'https://us02web.zoom.us/j/875384596',
+        'content': 'content',
+        'owner_id': user.id,
+    }
+
+    create_event(session, **data)
+
+    return user
+
+
+@pytest.fixture
+def event1(session, user2):
+    data = {
+        'title': 'test event title',
+        'start': datetime.strptime('2021-05-05 14:59', '%Y-%m-%d %H:%M'),
+        'end': datetime.strptime('2021-05-05 15:01', '%Y-%m-%d %H:%M'),
+        'location': 'https://us02web.zoom.us/j/87538459r6',
+        'content': 'content',
+        'owner_id': user2.id,
+    }
+
+    event = create_event(session, **data)
+    return event
+
+
+@pytest.fixture
+def user3(session, event1):
+    # a user made for testing who participates an event.
+    user = create_user(
+        session=session,
+        username='my_new_test_username2',
+        password='new_test_password2e',
+        email='new_t23est_love.email@gmail.com',
+        language_id='english'
+    )
+
+    association = UserEvent(
+        user_id=user.id,
+        event_id=event1.id
+    )
+    save(association, session)
+
+    return user
+
+
+# -----------------------------------------------------
+# Tests
+# -----------------------------------------------------
+
```

Comment:
Please don't use this comment style.

Reply:
why? 
i think it's very convenient; makes a nice seperation between fixtures and actual tests. seen this in other projects :)

Reply:
It's not currently a style we are using here, nor is it supported by PEP 8. Not sure we should start a new style now for this.

Reply:
deleted! :)